### PR TITLE
make golang http filter work with cgocheck

### PR DESCRIPTION
**Pull request by doujiang24**
so that people do not have to set `GODEBUG=cgocheck=0`.
fix https://github.com/envoyproxy/envoy/issues/25178

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: contrib/golang/filters/http/source/go/pkg/http/config.go [hunk 1]
```diff
@@ -67,11 +67,6 @@ func createConfig(c *C.httpConfig) *httpConfig {
 
 //export envoyGoFilterNewHttpPluginConfig
 func envoyGoFilterNewHttpPluginConfig(c *C.httpConfig) uint64 {
-	if !api.CgoCheckDisabled() {
```
Comment (spacewander) [inline contrib/golang/filters/http/source/go/pkg/http/config.go:70]: We can remove `CgoCheckDisabled` now?

**Review by spacewander — COMMENTED** [review]
Let's update the go.mod in example/ and test_data/

**Comment by doujiang24** [comment]
> Let's update the go.mod in example/ and test_data/

Thanks, updated the go.mod files related to golang HTTP extension.

**Review by spacewander — APPROVED** [review]

**Review by phlax — APPROVED** [review]
lgtm, thanks @doujiang24

**Pull request merged**